quipucords/api/source/serializer.py

```diff
@@ -529,10 +530,26 @@ class Meta:
             "ssl_cert_verify",
             "disable_ssl",
             "use_paramiko",
+            "proxy_url",
             "credentials",
             "most_recent_connect_scan",
         )
 
+    def validate_proxy_url(self, value):
+        """Validate that proxy URL is in the host:port format."""
+        if not value:
+            return value
+
+        SourceSerializerV2.is_valid_proxy_url_format(value)
+        return value
+
+    @staticmethod
+    def is_valid_proxy_url_format(proxy_url):
+        """Check that the proxy URL is in 'host:port' format."""
+        match = re.match(r"^([\w\.-]+):(\d+)$", proxy_url)
```

Comment:
Would we want to allow users not to specify the port number? Can we assume port 80/443 for plain/secure HTTP, or is that a false assumption?

i.e. Can a user just say "my proxy is https://my-great-proxy.local" without appending ":443"?

(Honest question; I don't know the typical user behavior here.)

Reply:
do we want only host:port to be specified? what about the protocol?

Reply:
Discussed elsewhere but recording it here for posterity: for the current iteration of this feature, we agreed simply to accept and require exactly `host:port` as the input for this single text field.